File: src/week1/ji/Baek_25705.py
n = int(input())
wheelArr = list(input().strip())
m = int(input())
targetArr = list(input().strip())

cnt = 0
idx = 0

while cnt <= 10000 and idx < len(targetArr):
    if targetArr[idx] == wheelArr[cnt % len(wheelArr)]:
        idx += 1
    cnt += 1

print(cnt if cnt <= 10000 else -1)

File: src/week1/ji/Baek_25705.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] wheelArr = sc.next().split("");
        int m = sc.nextInt();
        String[] targetArr = sc.next().split("");

        int cnt = -1;
        int idx = 0;

        while (cnt++ <= 10000 && idx < targetArr.length) {
            if (targetArr[idx].equals(wheelArr[cnt % wheelArr.length])) {
                idx++;
            }
        }

        System.out.println(cnt <= 10000 ? cnt : -1);
    }
}
